agent/consul/fsm/snapshot_oss.go

```diff
@@ -360,6 +364,27 @@ func (s *snapshot) persistIntentions(sink raft.SnapshotSink,
 	return nil
 }
 
+func (s *snapshot) persistConfigEntries(sink raft.SnapshotSink,
+	encoder *codec.Encoder) error {
+	entries, err := s.state.ConfigEntries()
+	if err != nil {
+		return err
+	}
+
+	for _, entry := range entries {
+		if _, err := sink.Write([]byte{byte(structs.ConfigEntryRequestType)}); err != nil {
+			return err
+		}
+		req := &structs.ConfigEntryRequest{
+			Entry: entry,
```

Comment:
Is relying on `0` Op value a good idea? I doubt we'd change constants around but if we did it might not be obvious we rely on the zero. Actually I guess even then it wouldn't break anything as long as the restore logic also ignores the Op name but still?

Reply:
Actually the logic for restore is just below and I see it ignores Op so it's not a big deal. I'd maybe add a comment here or both places that we are ignoring `Op` field and only using the `Request` struct for it's polymorphic encoding/decoding.